run_tool.sh

```diff
@@ -0,0 +1,92 @@
+#!/bin/bash
+#
+# A script to run the tool with all stages
+
+set -e
+
+
+# Ensure at least two positional parameters are provided
+if [ -z "$1" ] || [ -z "$2" ]; then
+  echo "Usage: $0 <tag1> <tag2> [-c clone_path] [-u repo_link] [-s subsystem]"
+  echo "error"
+  exit 1
+fi
+
+TAG1="$1"
+TAG2="$2"
+
+CLONE_PATH="linux-clone"
+REPO_URL="https://github.com/gregkh/linux"
+SUBSYS=""
+
+print_usage() {
+  echo "Usage: $0 <tag1> <tag2> [-c clone_path] [-u repo_link] [-s subsystem]"
+}
```

Comment:
It looks like this usage statement is duplicated earlier in this script. Would it be better to have the earlier function call this function? If so, you would want this function to appear earlier in the file so that it is available when it is needed.

Reply:
Thanks for pointing this out. I have moved this function earlier and used it in later code.